papertty/drivers/driver_it8951.py

```diff
@@ -357,15 +399,18 @@ def draw(self, x, y, image, update_mode_override=None):
             bpp = 4
             bpp_mode = self.BPP_4
 
-            #If the last write was in 1bpp mode, unset that register to take it out of 1bpp mode.
-            if self.in_bpp1_mode:
-                self.write_register(self.REG_UP1SR+2, self.read_register(self.REG_UP1SR+2) & ~(1<<2) )
-                self.in_bpp1_mode = False
+            #If this is the first (or only) image to draw, write the registers.
+            if isFirst:
+
+                #If the last write was in 1bpp mode, unset that register to take it out of 1bpp mode.
+                if self.in_bpp1_mode:
+                    self.write_register(self.REG_UP1SR+2, self.read_register(self.REG_UP1SR+2) & ~(1<<2) )
+                    self.in_bpp1_mode = False
 
-            #Then write the expected registers for 4bpp mode.
-            self.write_register(
-                    self.REG_MEMORY_CONV_LISAR + 2, (self.img_addr >> 16) & 0xFFFF)
-            self.write_register(self.REG_MEMORY_CONV_LISAR, self.img_addr & 0xFFFF)
+                #Then write the expected registers for 4bpp mode.
+                self.write_register(
+                        self.REG_MEMORY_CONV_LISAR + 2, (self.img_addr >> 16) & 0xFFFF)
+                self.write_register(self.REG_MEMORY_CONV_LISAR, self.img_addr & 0xFFFF)
```

Comment:
Indentation seems broken here

Reply:
Do you mean line 412? It was already indented like that prior to this PR. It's in brackets, so the extra indentation doesn't hurt.
If you mean both of those write_register statements, then the indentation is intentional. The registers only need to be written on the first chunk of data.

Reply:
Ah, right, misread the thing before having coffee :)